homeassistant/components/geniushub/__init__.py

```diff
@@ -106,18 +118,102 @@
 )
 
 
-async def async_setup(hass: HomeAssistant, config: ConfigType) -> bool:
+async def validate_input(
+    hass: HomeAssistant, hass_data_in: dict[str, Any]
+) -> dict[str, Any]:
+    """Validate the user input allows us to connect.
+
+    Data has the keys with values provided by the user.
+    """
+    hass_data = dict(hass_data_in)
+    mac = hass_data.pop(CONF_MAC, "")
+
+    if CONF_HOST in hass_data:
+        source = "Local: "
+        URL = "auth/release"
+        service = GeniusService(
+            hass_data.pop(CONF_HOST), **hass_data, session=async_get_clientsession(hass)
+        )
+    else:
+        source = "Cloud: "
+        URL = "version"
+        service = GeniusService(
+            hass_data.pop(CONF_TOKEN),
+            **hass_data,
+            session=async_get_clientsession(hass),
+        )
+
+    response = await service.request("GET", URL)
+
+    _LOGGER.debug("__init__.py:validate_input: ", extra=hass_data_in)
+    if source == "Local: ":
+        mac = response["data"]["UID"]
+
+    return {"title": source + mac}
+
+
+async def _async_import(hass: HomeAssistant, base_config: ConfigType) -> None:
+    """Import a config entry from configuration.yaml."""
+
+    result = await hass.config_entries.flow.async_init(
+        DOMAIN,
+        context={"source": config_entries.SOURCE_IMPORT},
+        data=base_config[DOMAIN],
+    )
+    if result["type"] == FlowResultType.CREATE_ENTRY:
```

Comment:
There needs to be an exception for the `already_exists` abort

Reply:
Happy for you to write the mock tests. Over to you.